src/pocketmine/level/Level.php

```diff
@@ -1505,6 +1506,42 @@ public function dropItem(Vector3 $source, Item $item, Vector3 $motion = null, in
 	}
 
 	/**
+	 * @param Vector3 $pos
+	 * @param string  $text
+	 * @param string  $title
+	 *
+	 * @return null|Entity
+	 */
+	public function addFloatingText(Vector3 $pos, string $text, string $title = ""){
+		$entity = Entity::createEntity("FloatingText", $this, new CompoundTag("", [
+			new ListTag("Pos", [
+				new DoubleTag("", $pos->x),
+				new DoubleTag("", $pos->y),
+				new DoubleTag("", $pos->z)
+			]),
+			new ListTag("Motion", [
+				new DoubleTag("", 0),
+				new DoubleTag("", 0),
+				new DoubleTag("", 0)
+			]),
+			new ListTag("Rotation", [
+				new FloatTag("", 0),
+				new FloatTag("", 0)
+			])
+		]));
+
+		assert($entity !== null);
+
+		if($entity instanceof FloatingText){
```

Comment:
Why is this not an assertion?

Reply:
plugins might override it with their own entities